jbmc/unit/java_bytecode/java_trace_validation/java_trace_validation.cpp

```diff
@@ -0,0 +1,128 @@
+/*******************************************************************\
+
+Module: Unit tests for java_trace_validation
+
+Author: Diffblue Ltd.
+
+\*******************************************************************/
+
+#include <goto-programs/goto_trace.h>
+#include <java_bytecode/java_trace_validation.h>
+#include <java_bytecode/java_types.h>
+#include <testing-utils/use_catch.h>
+#include <util/byte_operators.h>
+
+TEST_CASE("java trace validation", "[core][java_trace_validation]")
+{
+  const exprt plain_expr = exprt();
+  const exprt expr_with_data = exprt("id", java_int_type());
+  const symbol_exprt valid_symbol_expr = symbol_exprt("id", java_int_type());
+  const symbol_exprt invalid_symbol_expr = symbol_exprt(java_int_type());
```

Comment:
:bulb: To be complete, you might consider some other invalid symbol exprs (ones with subexpressions for example)